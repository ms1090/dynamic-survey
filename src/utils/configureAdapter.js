import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// configure adapter to use enzyme in unit test cases
configure({ adapter: new Adapter() });
