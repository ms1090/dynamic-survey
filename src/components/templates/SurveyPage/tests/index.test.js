import React from 'react';
import { shallow } from 'enzyme';
import { SurveyPageVanilla } from '..';
import toJson from 'enzyme-to-json';
import 'utils/configureAdapter';

describe('SurveyPageVanilla Component', () => {
  const fetchQuestionsMock = jest.fn();
  const fetchUserAnswersMock = jest.fn();
  beforeEach(() => {

  });
  test('should render correctly', () => {
    let wrapper = shallow(<SurveyPageVanilla
      quesCount={4}
      currQuesIndex={1}
      fetchQuestions={() => {}}
      fetchUserAnswers={() => {}}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper = shallow(<SurveyPageVanilla
      quesCount={4}
      currQuesIndex={4}
      fetchQuestions={() => {}}
      fetchUserAnswers={() => {}}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test('should call fetchQuestions, fetchUserAnswers after mount', () => {
    shallow(<SurveyPageVanilla
      quesCount={4}
      currQuesIndex={1}
      fetchQuestions={fetchQuestionsMock}
      fetchUserAnswers={fetchUserAnswersMock}
    />);
    expect(fetchQuestionsMock.mock.calls.length).toBe(1);
    expect(fetchUserAnswersMock.mock.calls.length).toBe(1);
  });
});
