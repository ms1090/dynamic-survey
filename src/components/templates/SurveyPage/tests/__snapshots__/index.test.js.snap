// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SurveyPageVanilla Component should render correctly 1`] = `
<div
  className="layout-container"
>
  <ProgressBar
    className=""
    max={4}
    value={1}
  />
  <Connect(QuestionForm) />
</div>
`;

exports[`SurveyPageVanilla Component should render correctly 2`] = `
<div
  className="layout-container"
>
  <ProgressBar
    className=""
    max={4}
    value={4}
  />
  <Connect(SurveySummary) />
</div>
`;
